начал

Jhon-Crow opened the pull request:
есть лого)

jhamish committed:
начал
есть лого)

jhamish committed:
добавил нав бар

Jhon-Crow reviewed (commented):
добавил нав бар

jhamish committed:
header-line выглядит готовым

jhamish committed:
начал разметку 1 страницы

jhamish committed:
добавил картинку и спаны

jhamish committed:
пытаюсь выравнять

jhamish committed:
пытаюсь выравнять 2

jhamish committed:
сделал display: flex
выравниваю

jhamish committed:
первый контейнер почти готов

осталось сделать векторные кривые на бэкграунд

jhamish committed:
начал 2 контейнер

jhamish committed:
переделываю, не получается добавить картинку на background

jhamish committed:
победил background

jhamish committed:
1 секция полностью готова

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+    margin: 0 1rem;
+}
+
+h3 {
```
на теги стили не вешаем

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+
+h3 {
+    margin: 0;
+    color: #FFF;
```
цвета в пременные

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    /*flex-wrap: wrap;*/
```
код не Комментируем, до пул реквеста это доходить не должно

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+}
+
+
+.nav2 {
```
классам давай более осознанное название

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+
+
+.nav2 {
+    margin-left: 5%;
```
старайся марджины тоже в Ремах указывать а не в процентах, иначе развалится на адптивной верстке

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+    justify-content: center;
+    /*flex-wrap: wrap;*/
+    border-radius: 5.625rem;
+    background: radial-gradient(2900.76% 50.13% at 50.00% 53.89%, rgba(58, 52, 86, 0.95) 0%, #211E2E 100%);
```
цвета в переменные

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+}
+
+.background1 {
+    background: url(" /Programming/assets/1-base-image.png");
```
картинки в переменные

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+    /*padding: 17px 38px 16px 34px;*/
+    padding: 1.06rem 2.375rem 1rem 2.125rem;
+    color: #343045;
+    background: linear-gradient(90deg, #8176AF 0%, #C0B7E8 100%);
```
пройдись по всему коду перенеси все цвета в переменные

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+    padding-left: 0.94rem;
+}
+
+input[type='button'] {
```
если это не сброс стилей, а это вроде не он, то добавь эти стили в класс и класс навесь на инпут

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+    list-style: none;
+    display: flex;
+    color: #FFF;
+    font-size: 0.75rem;;
```
Проверь код на печатки, тут две ;;

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+    position: fixed;
+    left: 0;
+    top: 0;
+    min-width: 90%;
```
если нужны отступы лучше сделай через падинги

jegius replied:
старайся чтобы размер определялся контентом

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+    line-height: normal;
+}
+
+p {
```
на теги стили не вешаем

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+
+}
+
+input {
```
кроме сбросо

jegius commented on css/main.css:
```diff
@@ -0,0 +1,241 @@
+    padding: 0;
+    font-family: Montserrat, sans-serif;
+    background-color: rgb(48, 44, 66);
+    background-image: url(" /Programming/assets/vector-all.png");
```
суверен что картинка нужна на бэкграунда боди а не на секции?

Jhon-Crow replied:
победил
и правда на секцию надо было

jhamish committed:
поправил всё кроме бекграунда

jhamish committed:
победил бекграунд, первая секция готова

jhamish committed:
начал 2 cart

jhamish committed:
начал 3 секцию, карточки service

jhamish committed:
начал делать process-section (пока не понятно как делать)

jhamish committed:
сделал process-section

jhamish committed:
всё готово

jhamish committed:
подчищено и ссылки на соцсети теперь <a>

jhamish committed:
переделал поля ввода

jhamish committed:
дочистил

Jhon-Crow commented:
отревьювте пжпж

jhamish committed:
убрал <br>

Yashiandr commented on index.html:
```diff
@@ -1,10 +1,246 @@
+        <div class="header-button-wraper">
+    <input class="header-contact-button" type="button" value="CONTACT US">
+        <input class="header-join-button light-button" type="button" value="JOIN HYDRA">
+        </div>
```
Отформатируй код, должна быть лесенка в html, а не вот это вот

Yashiandr commented on index.html:
```diff
@@ -1,10 +1,246 @@
+        </div>
+        <div class="header-logo-text">
+            <img src="/assets/logo-text.png" alt="logo">
+        </div>
```
Картинки вставляй в бэкграунд в ксс, а не в html.

Yashiandr replied:
И когда будешь вставлять картинки в `background`, то не забудь их указывать в переменные, также как и цвета

Yashiandr commented on index.html:
```diff
@@ -1,10 +1,246 @@
+        <div class="cart _with-padding">
+            <h1 class="biggest-title"><span class="header-title-span">Dive</span> Into The Depths</h1>
+            <h1 class="biggest-title">Of <span class="header-title-span">Virtual Reality</span></h1>
+            <p class="section-main-text _height-padding block1_p">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore nisl tincidunt eget. Lectus mauris eros in vitae .</p>
```
отформатируй код, не просто так же prettier поставил. Максимальная длина строки кода равна 80 символов

Yashiandr commented on index.html:
```diff
@@ -1,10 +1,246 @@
+                <img class="arrow-small-right" src="/assets/arrow-small-right.png" alt="arrow-small-right">
+        </div>
+        </div>
+        <address class="address">
+            <div class="address-item">
+                <img src="/assets/Location-Icon.png" alt="Location-Icon">
+            </div>
+            <div class="address-item">
+                <h3 class="small-section-title">Pay Us a Visit</h3>
+                <h4 class="address-main-text">Union St, Seattle, WA 98101, United States</h4>
+            </div>
+            <div class="address-item">
+                <img src="/assets/Vector%205.png" alt="Vector 5">
+            </div>
+            <div class="address-item">
+                <img src="/assets/phone-call.png" alt="phone-call">
+            </div>
+            <div class="address-item">
+                <h3 class="small-section-title">Give Us a Call</h3>
+                <h4 class="address-main-text">(110) 1111-1010</h4>
+            </div>
+            <div class="address-item">
+                <img src="/assets/Vector%205.png" alt="Vector 5">
+            </div>
+            <div class="address-item">
+                <img src="/assets/mail.png" alt="mail">
+            </div>
+            <div class="address-item">
+                <h3 class="small-section-title">Send Us a Message</h3>
+                <h4 class="address-main-text">Contact@HydraVTech.com</h4>
+            </div>
+        </address>
```
Это ненумерованный список (`<ul>`)

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,636 @@
+    backdrop-filter: blur(0.063rem);
+    z-index: 1;
+    position: fixed;
+    left: 0;
+    top: 0;
```
лишнее

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,636 @@
+.header {
+    backdrop-filter: blur(0.063rem);
+    z-index: 1;
+    position: fixed;
```
если хочешь сделать фиксированный хедер, то тогда необходимо растянуть его на всю ширину (`width: 100%`) и чтобы он не заходил за границы экрана поставить другой `justify-content`

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,636 @@
+}
+
+.nav {
+    list-style: none;
```
дублируешь стиль сброса, указанный тобой выше

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,636 @@
+    font-style: normal;
+    font-weight: 700;
+    line-height: normal;
+    margin-left: 9.81rem;
```
здесь можно обойтись без margin, старайся, чтобы расположение определялось свойствами flex'а, а не марджинами, марджины в крайнем случае используем, то бишь когда дизайнер сделал кривой дизайн, когда всё симметрично, то можно обойтись без них

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,636 @@
+    list-style: none;
+    display: flex;
+    color: var(--white);
+    font-size: 0.75rem;
```
размеры шрифтов указываем в px, остальное в rem

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,636 @@
+}
+
+.header-logo-img {
+    filter: drop-shadow(0 0.625rem 0.375rem rgba(0, 0, 0, 0.25));
+}
+
+.header-logo-text {
+    padding-left: 0.94rem;
```
у чисел с нулем в целой части 0 не указывает 0.25 -> .25

Jhon-Crow replied:
готово

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,636 @@
+}
+
+.tech-section-back-image {
+    background-image: url(" /Programming/assets/pexels-rodnae-productions-8097332.png");
+    background-size: 79.9rem 18.9rem;
+    background-position: center;
+    background-repeat: no-repeat;
```
можно объединить в одно свойство background. Также изображения указываем в переменные.

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,636 @@
+.tech-image1 {
+    width: 10.875rem;
+    height: 10.875rem;
+    flex-shrink: 0;
```
flex-shrink в данном и последующем случае не значит ничего, не копируй бездумно из фигмы свойства

Jhon-Crow replied:
готово

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,636 @@
+    overflow: hidden;
+}
+
+.circle::before {
+    content: '';
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    background: linear-gradient(156deg, var(--lightest-violet) 0%, var(--lighter-violet) 100%);
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    color: var(--dark-violet);
+    font-size: 4rem;
+    font-weight: 700;
+}
+
+.circle1::before {
+    content: '01';
+}
+
+.circle2::before {
+    content: '02';
+}
+
+.circle3::before {
+    content: '03';
+}
+
+.circle4::before {
+    content: '04';
+}
```
Кажется, как будто можно обойтись без псевдоэлемента, указав круг и границу его в одном классе, а текст в html

Jhon-Crow replied:
пока не знаю как избежать побочного эффекта
![image](https://github.com/jegius/ivan-edu/assets/133867474/3aff14be-7627-4731-ae7a-edb4f57d3391)

Jhon-Crow replied:
сделал картинку на фон+текст в html

Jhon-Crow replied:
сделал картинкой

Yashiandr reviewed (commented):
Ряд общих, но очень важных замечаний:

1) Картинки храним внутри папки проекта в папке image и ссылаемся на них в ней, я сейчас смотрел верстку вообще без картинок, ибо их просто нету на гитхабе. Также картинки складываются не общей кучей, а сортируются в папки по блокам.
2) Названия классов не соответствует методологии БЭМ, не прослеживается блочно-элементная структура, вижу только модификаторы, переименуй все классы согласно БЭМ.
3) Не все цвета указаны в переменных, туда необходимо указывать и градиенты, и цвета, указанные с помощью функции rgba. Также и изображения указываем в переменные
4) Отформатируй код!!!!
5) Нули у чисел с 0 в целой части 0 не указывается.
6) Старайся, чтобы расположение контента прежде всего шло от flex'а, а не от марджинов, чтобы сделать расстояние между элементов одного контейнера лучше использовать `gap`.

И ещё раз повторю, используй БЭМ при наименовании классов, вернее переименуй их всех согласно БЭМ и поставь в своем редакторе форматирование при сохранении.

jhamish committed:
image

jhamish committed:
image  пока в кучу

jhamish committed:
переворашиваю всё
прогресс ~ 10%

jhamish committed:
надеюсь то делаю
переименовываю классы

jhamish committed:
переименовал классы

jhamish committed:
разложил картинки по папкам
осталось:
1)убрать в переменные
2)перенести на бекграунд

jhamish committed:
1)перенести на бекграунд (~50% done)
2)убрать в переменные

jhamish committed:
1)перенести на бекграунд (~60% done)
2)убрать в переменные

jhamish committed:
1)перенести на бекграунд (~70% done)
2)убрать в переменные (всё что на бекграунде уже в переменных)

jhamish committed:
с картинками закончил
осталось: переделать process-line

jhamish committed:
переделал process-line
осталось:
1)градиенты в переменные
2)сделать address ненумированным списком

jhamish committed:
всё сделал
что дальше не знаю)

jhamish committed:
заменил /Programming/image
на ../image

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,744 @@
+.header__contact-button {
+    border: .125rem solid var(--white);
+    color: var(--white);
+    margin-left: 8.75rem;
```
Много какие марджины не нужны, ибо у тебя у родительского класса стоит justify-content, который их расставляет нормально по местам. Я например удалил этот и ничего не поменялось, ещё такие на мой взгляд остались, поубирай лишнее.

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,744 @@
+    filter: drop-shadow(0 .625rem .375rem var(--shadow));
+}
+
+.button {
+    border-radius: 2.5rem;
+    cursor: pointer;
+    padding: 1rem 2.1rem;
+    box-shadow: 0 0 0 0;
+    transform: scale(1);
+    transition: transform .2s ease-in-out, box-shadow .2s ease-in-out;
+}
+
+.button:hover {
+    transform: scale(.97);
+    box-shadow: 0 0 .95rem var(--button-hover);
+}
```
так как ты данный стиль используешь много где, а не только в одном месте, то логичнее его разместить отдельно от всего кода, либо сверху, либо снизу, чтобы не искать его каждый раз, если не в хедере решил посмотреть на кнопку.

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,744 @@
+    margin-left: 2.38rem;
+}
+
+.hero-section {
+    padding-top: 18rem;
+}
```
Конечно по-хорошему, чтобы флекс верстка расставляла все объекты по их местам, но это тебе всё переписывать придется, я пока принимаю, но Иван может придраться к такому способу расположения контента.

Yashiandr replied:
Нет, всё-таки не принимаю, надо переделывать.

Jhon-Crow replied:
Вот с этим местом вообще не знаю как быть

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,744 @@
+    gap: 0 2.66rem;
+}
+
+.address-item_1 {
+    background: var(--address-item1);
+    height: 4.375rem;
+    width: 4.375rem;
+}
+
+.address-item_2 {
+    background: var(--address-item2);
+    width: 3.6875rem;
+    height: 3.6875rem;
+}
+
+.address-item_3 {
+    background: var(--address-item3);
+    width: 4.0625rem;
+    height: 4.0625rem;
+}
+
+.address-vector {
+    background: var(--address-vector);
+    width: .0625rem;
+    height: 7.28125rem;
+}
```
no-repeat поставь к бэкграунду.

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,744 @@
+    margin-top: 1rem;
+}
+
+._on-right-side {
```
Забавно, у тебя вот этот класс используется один раз для класса, в котором написан padding-top и padding-bot (about-section__cart), лишь бы не одним педдингом написать в одном классе =D. Но вообще такая тема должна делаться flex версткой или хотя бы display: inline, а вот это всё перенос на правую сторону объекта с помощью пэддингов

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,744 @@
+    height: 4.1875rem;
+}
+
+._margin-bottom-7 {
+    margin-bottom: 7.25rem;
+}
```
ужас, избавься от этого

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,744 @@
+    color: var(--dark-violet);
+}
+
+.process-section__line {
+    margin-top: 2rem;
+    padding: 0 3.81rem;
+    display: flex;
+    justify-content: space-between;
+}
+
+.process-section__line2 {
+    padding: 0 1.81rem;
+    display: flex;
+    justify-content: space-between;
+    align-items: flex-start;
+}
```
не используй в названиях цифры, некрасиво это, давай имена согласно свойствам объекта.

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,744 @@
+    border-radius: 10rem;
+}
+
+.soc-item1 {
+    background: var(--facebook) no-repeat;
+    background-size: 2rem;
+}
+
+.soc-item2 {
+    background: var(--twitter) no-repeat;
+    background-size: 2rem;
+}
+
+.soc-item3 {
+    background: var(--linkedin) no-repeat;
+    background-size: 2rem;
+}
+
+.soc-item4 {
+    background: var(--youtube) no-repeat;
+    background-size: 2rem;
+}
+
+.soc-item5 {
+    background: var(--instagram) no-repeat;
+    background-size: 2rem;
+}
+
+.soc-item6 {
+    background: var(--pinterest) no-repeat;
+    background-size: 2rem;
```
Логичнее назвать: 
.soc-item__facebook
.soc-item__twitter.
и тд.

Если бы ты картинки не назвал соответствующе, я бы вообще не понял за что это отвечает всё.

Yashiandr reviewed (commented):
Так, создай для каждой секции свой блок в ксс (родительский стиль), а не только запихивай в класс секции изображения на бэкграунд. Этому родительскому стилю секции присвой display: flex и расставляй контент с помощью свойств flex и свойства gap, избавься от paddingов и marginов, у тебя это возможно. Вот шпаргалка:
![image](https://github.com/jegius/ivan-edu/assets/115163286/4ced6362-6e64-4b1e-9c4b-06b83df188de)
также уже можешь заменять width на min-width или max-width, кроме всяких маленьких объектов и псевдоэлментов.
Всему того, что "ссылка" присвой анимацию какую-либо, чтобы было понятно, что навел на ссылку, это в футере у тебя.
Не используй цифры в названиях классов, называй согласно их контексту и свойствам. цифры не несут в себе четкую информацию.
И вообще идеально было бы, если бы ты упорядочил стили в css-файле по формату:
1)переменные
2)стили сброса
3)миксины (общие для всего файла стили, такие как button, который ты везде используешь, но находится он только в header), ну и их можно указать и в самом низу, главное чтобы они все были отдельно от основных стилей.
4)Тут основные стили которые идут сверху вниз также, как вызываются сверху вниз в html, то бишь упорядоченно и так, что легче в них будет ориентироваться

jhamish committed:
переименовал айтемны

jhamish committed:
переделываю секции
1 готова

jhamish committed:
исправиль почти всё

jhamish committed:
header теперь position: sticky;
поправил что съехало

jhamish committed:
убрал марджин из хедера

jhamish committed:
поменял размер about-section__image_face

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+    line-height: 1.75rem;
+}
+
+.item__title_small {
```
По методологии БЭМ нету модификатора элемента блока, то бишь у тебя либо модификатор блока, либо элемент блока, либо миксин (модификатор вне блока, использующийся во всех местах кода, а не только в одном блоке), так что переименуй либо чтобы это был либо элемент, либо модификатор.

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+    line-height: normal;
+}
+
+.section__title_slim {
```
Элемент это или модификатор?

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+    line-height: normal;
+}
+
+.section__title_regular {
```
Это элемент или модификатор?

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+    font-weight: 700;
+}
+
+._margin-top_1rem {
```
это модификатор модификатора?

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+    list-style-type: none;
+}
+
+/*Mixы*/
```
Вообще, если это миксин, то его лучше называть начиная с _, чтобы не возникало путаницы, блок это или что ещё  в самой html не понятно из-за этого. Ну и у миксов нет как таковых элементов или модификаторов, они сами по себе модификаторы стиля.

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+/*Mixы*/
+.section__main-text {
+    color: var(--white);
+    font-size: 1rem;
```
шрифты в px, всё остальное, кроме font-size, в rem

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+
+.item__title_small {
+    color: var(--white);
+    font-size: 1.5rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+
+.section__title_slim {
+    white-space: nowrap;
+    font-size: 2.25rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+}
+
+.section__title_regular {
+    font-size: 2.25rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+    cursor: pointer;
+    padding: 1rem 2.1rem;
+    box-shadow: 0 0 0 0;
+    transform: scale(1);
```
данный трансформ не делает ничего, можно оставить просто в ховере и всё

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+    border-radius: 2.5rem;
+    cursor: pointer;
+    padding: 1rem 2.1rem;
+    box-shadow: 0 0 0 0;
```
данный бокс-шадоу не делает ничего, оставь просто в ховере

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+.header__nav {
+    display: flex;
+    color: var(--white);
+    font-size: .75rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+
+.header__nav {
+    display: flex;
+    color: var(--white);
```
ещё чего заметил, у тебя цвет, за исключением кнопки и пары момент в начале, везде белый, и в main-font, классе родительского блока, указан белый цвет шрифта, это свойство не делает ничего и можно её убрать, как и везде ниже.

jegius replied:
чеки эти комментарии, действительно выглядит что тот же цвет можно в одном месте указать, проверь остальные прочерти

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+
+.header__nav-item {
+    padding: 2.75rem 1.25rem;
+    color: var(--white);
```
ничего не делает, ну и тд, посмотри сам. полно тут лишнего

Yashiandr replied:
![image](https://github.com/jegius/ivan-edu/assets/115163286/1061a7ac-27a7-4df9-a69d-0a19da810851)
![image](https://github.com/jegius/ivan-edu/assets/115163286/ba9e6703-aa76-440f-b9d3-f0ac9cbf1613)
![image](https://github.com/jegius/ivan-edu/assets/115163286/95d2f4ca-2136-47bc-8462-269d5c07bf58)
у тебя в трех местах указан белый цвет для header__nav-item, что можно наглядно увидеть через f12 на страничке своей. Также много где

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+    border: none;
+}
+
+.service-section__main-text_smallest {
```
Элемент или модификатор?

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+    width: 6.25rem;
+    height: 6.25rem;
+    border-radius: 10rem;
+    box-shadow: 0 0 0 0;
+    transform: scale(1);
```
данные строчки ничего не делают

Jhon-Crow replied:
для анимации добавил, чтоб обратно тоже отыгрывалась, если правильно понял

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+.optin-section__pulls-border::placeholder {
+    color: var(--white);
+    font-family: Montserrat, sans-serif;
+    font-size: .875rem;
```
шрифты в пикселях

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+    line-height: normal;
+}
+
+._small-pull {
+    width: 30.13rem;
+    height: 4.5rem;
+}
+
+._big-pull {
+    width: 63.125rem;
+    height: 4.5rem;
+}
+
+._textarea {
+    width: 66.125rem;
+    height: 13.125rem;
+    padding-left: 2.62rem;
+    padding-top: 1.69rem;
+}
```
Если это миксины, то их ко всем отнеси

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+.soc-item {
+    width: 2rem;
+    height: 2rem;
+    box-shadow: 0 0 0 0;
+    transform: scale(1);
```
ничего не делают

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,775 @@
+}
+
+.credits-text {
+    font-size: .875rem;
```
шрифты в пикселях

Yashiandr reviewed (commented):
В целом очень неплохо, прочитай побольше про БЭМ, переименуй согласно нему классы, чтобы у всего что элемент был блок и так далее там по коду ещё замечания устрани, и можешь ещё заменять width у всего, кроме незначительных маленьких элементов на min-width или max-width в зависимости от контекста. Такими темпами скоро скажу Ивану проверить уже.

jhamish committed:
нич-чего не понимаю, начал переписывать названия по бэм, всё ещё не уверен что делаю то
шрифты переведены в px
дописал хедеру гап
подчистил

jhamish committed:
шрифты переведены в px

jhamish committed:
переписал в бэм

jhamish committed:
подчистил

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,758 @@
+    font-weight: 300;
+}
+
+.section__p {
```
назови хотя бы paragraph или text, а не __p

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,758 @@
+    line-height: 3rem;
+}
+
+.footer__a {
+    box-shadow: 0 0 0 0;
+    transition: box-shadow .3s ease-in-out;
+}
+
+.footer__a:hover {
```
Этому ещё тоже адекватное название

Jhon-Crow replied:
footer__link

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,758 @@
+    background-size: 2rem;
+}
+
+.footer__soc-item_pinterest {
```
не по бэм название

Jhon-Crow replied:
footer__soc-item-pinterest

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,758 @@
+    background-size: 2rem;
+}
+
+.footer__soc-item_twitter {
+    background: var(--twitter) no-repeat;
+    background-size: 2rem;
+}
+
+.footer__soc-item_linkedin {
+    background: var(--linkedin) no-repeat;
+    background-size: 2rem;
+}
+
+.footer__soc-item_youtube {
+    background: var(--youtube) no-repeat;
+    background-size: 2rem;
+}
+
+.footer__soc-item_instagram {
+    background: var(--instagram) no-repeat;
+    background-size: 2rem;
+}
```
не по бэм

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,758 @@
+    box-shadow: 0 0 .95rem var(--button-hover);
+}
+
+.section__line_tech {
```
не по бэм

Jhon-Crow replied:
tech-section__line

Yashiandr reviewed (commented):
НУ поправь незначительные замечания и замени width и height на min(или max)-width/height везде, кроме маленьких элементов, и тогда отправлю Ивану на ревью

jhamish committed:
переписал height width на max min

jhamish committed:
исправил по указаниям

jegius commented on index.html:
```diff
@@ -1,10 +1,305 @@
-  <title>$Title$</title>
+	<meta charset='UTF-8'>
+	<title>Hydra Landing Page</title>
+	<link href='/css/main.css' rel='stylesheet'>
```
неправильно указаны пути к main.css, отсюда проблема с подгрузкой картинок, путь должен быть относительным `./css/main.css`

jegius commented on index.html:
```diff
@@ -1,10 +1,305 @@
-<body>
-$END$
+<body class='main__background main-font'>
+<main class='main vector-image-for-hero-section-right'>
```
класс main нужно ограничеть по ширине чтобы он корректно отображался на широких мониторах

jegius commented on index.html:
```diff
@@ -1,10 +1,305 @@
+				<div class='process-section__circle'>03</div>
+				<div class='process-section__circle'>04</div>
+			</div>
+			<div class='process-section__text-line'>
```
это явно список

jegius commented on index.html:
```diff
@@ -1,10 +1,305 @@
+				magna fringilla urna porttitor rhoncus vitae.</p>
+		</div>
+		<div>
+			<div class='process-section__circle-line'>
```
и это тоже список

jegius commented on css/main.css:
```diff
@@ -0,0 +1,760 @@
+    --optin-gradient: radial-gradient(50% 50% at 50% 50%, var(--violet) 0%, var(--darkest-violet) 100%);
+
+    /*images*/
+    --logo-icon: url(" /Programming/image/logo.png");
```
пробелы из путей убери везде и исправь импорт стилей в index.html потом везде сделай относительные пути. `/Programming/image/logo.png` => `../image/logo.png`

jegius commented on css/main.css:
```diff
@@ -0,0 +1,760 @@
+@import url('https://fonts.googleapis.com/css2?family=Montserrat&display=swap');
+
+:root {
+    /*color*/
```
все эти коментарии не имею смысла, убери их отовсюду

jegius commented:
![image](https://github.com/jegius/ivan-edu/assets/11362702/fad0c1bf-96c4-4816-ad8a-0a77dbb3808d)
съехал пункт, поправь

jegius commented:
![image](https://github.com/jegius/ivan-edu/assets/11362702/71913244-9da6-4f7b-b7d6-5d6f35f524ad)
проверь выравнивание

jhamish committed:
исправил и почистил по указаниям

jegius commented on css/main.css:
```diff
@@ -0,0 +1,745 @@
+    padding-top: 3.12rem;
+    max-width: 100%;
+    backdrop-filter: blur(.063rem);
+    z-index: 1;
```
тебе тут z-index не нужен

Jhon-Crow replied:
без него кнопки поверх хедера лезут при прокрутке

jegius commented on index.html:
```diff
@@ -1,10 +1,317 @@
+				<div class='footer__line-vector-border'></div>
+				<div class='footer__socialize'>
+					<h3>SOCIALIZE WITH HYDRA</h3>
+					<div class='footer__soc-wrapper'>
```
это тоже список

jegius commented on index.html:
```diff
@@ -1,10 +1,317 @@
+			<div class='hero-section__arrow'></div>
+			</div>
+		</div>
+			<div class='hero-section__image-face'></div>
```
проверь форматирование, че оно. у тебя по файлу хромает, юзай хоткей для автоформатирования

jhamish committed:
форматирование + список soc-items

jegius commented on index.html:
```diff
@@ -1,10 +1,346 @@
+		</div>
+		<address class='hero-section__address'>
+			<div class='address-item-location'></div>
+			<ul>
```
члушай тут наверное все таки не список, сделай через заголовок h5 + span

jegius commented on index.html:
```diff
@@ -1,10 +1,346 @@
+			</div>
+			<div class='hero-section__image-face'></div>
+		</div>
+		<address class='hero-section__address'>
```
думаю не зачем использовать тег адресс

jegius commented on index.html:
```diff
@@ -1,10 +1,346 @@
+					class='hero-section__span hero-section__span_grad-92deg'>Dive</span>
+					Into The Depths</h1>
+				<h1 class='section__title section__title_hero'>Of <span
+					class='hero-section__span hero-section__span_grad-92deg'>Virtual Reality</span>
```
слушай а зачем тут спан? почему эти классы нельзя на заголовок навесить?

Jhon-Crow replied:
если сделать заголовок то будет с новой строки, а тут надо в одну строку но пару слов перекрасить в градиент

Jhon-Crow replied:
нужно чтоб в одной строке и раскрашенные градиентом и обычные буквы были

Yashiandr replied:
ну да, я также сделал и ко мне претензий не было, а как иначе сделать я не знаю

jegius commented on index.html:
```diff
@@ -1,10 +1,346 @@
+			<div
+				class='service-section__item service-image_SIMULATION'>
+				<h3 class='item-title'>SIMULATION</h3>
+				<hr class='service-section__hr'>
```
hr, br не юзаем

jegius commented on index.html:
```diff
@@ -1,10 +1,346 @@
+			</div>
+			<div class='service-section__item service-image_SELF-CARE'>
+				<h3 class='item-title'>SELF-CARE</h3>
+				<hr class='service-section__hr'>
```
тут и ниже тоже

jegius commented on index.html:
```diff
@@ -1,10 +1,346 @@
+	</section>
+	<section class='section process-section__back-image'>
+		<div class='section__line'>
+			<div>
```
зачем тебе тут пустой див?

Jhon-Crow replied:
тут вроде всё полное. пустой только div class='arrow-big' пустой, но он - картинка
или о чём тут речь?

Jhon-Crow replied:
так вроде не пустой

Yashiandr replied:
пустой в смысле класс не задан никакой, наверное это имел в виду Иван.

Jhon-Crow replied:
этот див чтоб в линию элементы были

jegius commented on index.html:
```diff
@@ -1,10 +1,346 @@
+		</div>
+		<ul class='process-section__circle-line'>
+			<li>
+				<div class='process-section__circle'>01</div>
```
вот это все оберни в карточки

Jhon-Crow replied:
тут не понял что как
обернул всю эту часть с кругами

jhamish committed:
исправил по мелочи

Yashiandr commented on css/main.css:
```diff
@@ -0,0 +1,747 @@
+}
+
+.footer__link {
+    box-shadow: 0 0 0 0;
```
задай цвет box-shadow здесь, так как по умолчанию он белый и из-за этого не самый приятный hover, ибо при наведении внезапно появляется белый боксшадоу и меняет свой цвет по анимации до --darkest-violet за 30 миллисекунд, что ну такое себе

Jhon-Crow replied:
забавный ховер был)
поправил

Yashiandr commented on index.html:
```diff
@@ -1,10 +1,348 @@
+			</div>
+		</div>
+	</section>
+	<section class='section process-section__back-image'>
```
Вообще по-хорошему этим секциям надо задать flex-wrap: wrap, чтобы при приближении не текст съеживался до одного символа в длину, а переносился под заголовок секции. Так как мы в дальнейшем эту верстку размещаем ещё и на хостинге гитхаба и в теории эту верстку можно будет посмотреть и с телефона, то в телефоне она будет выглядеть вот как-то так.
![image](https://github.com/jegius/ivan-edu/assets/115163286/00e8384b-a158-48dc-bcd9-178ffb1b6da1)

Yashiandr replied:
надо бы тоже это поправить и для кнопок выше, всё это можно разными путями сделать, но основной это всему задать возможность переноса с помощью flex-wrap: wrap, но скорее всего подрефакторить код придется ради этого, расстояния выписать, структуру в html, короче необходимый гемор.

Jhon-Crow replied:
тут всё в столбик надо?

Yashiandr commented on index.html:
```diff
@@ -1,10 +1,348 @@
+		<div class='optin-section'>
+			<h1 class='section__title'>JOIN HYDRA</h1>
+			<div class='optin-section__vector-hr'></div>
+			<h2 class='section__title section__title_slim'>Let’s Build Your VR
+																		  Experience</h2>
```
О_о

Yashiandr commented on index.html:
```diff
@@ -1,10 +1,348 @@
+					<h3>SOCIALIZE WITH HYDRA</h3>
+					<ul class='footer__soc-wrapper'>
+						<li>
+							<a href=''>
+								<div
+									class='footer__soc-item footer__soc-item-facebook'></div>
+							</a>
+						</li>
+						<li>
+							<a href=''>
+								<div
+									class='footer__soc-item footer__soc-item-twitter'></div>
+							</a>
+						</li>
+						<li>
+							<a href=''>
+								<div
+									class='footer__soc-item footer__soc-item-linkedin'></div>
+							</a>
+						</li>
+						<li>
+							<a href=''>
+								<div
+									class='footer__soc-item footer__soc-item-youtube'></div>
+							</a>
+						</li>
+						<li>
+							<a href=''>
+								<div
+									class='footer__soc-item footer__soc-item-instagram'></div>
+							</a>
+						</li>
+						<li>
+							<a href=''>
+								<div
+									class='footer__soc-item footer__soc-item-pinterest'></div>
+							</a>
+						</li>
+					</ul>
```
выглядит страшно, проверь настройки форматирования, поиграйся с ними

jhamish committed:
исправил по мелочи
что делать с форматирование не знаю
надо будет для телефонного формата сделать flex-wrap

jhamish committed:
попытался сделать flex-wrap

jhamish committed:
переделала текстовые поля, теперь не вылезают за границы на маленьком экране

jhamish committed:
добил flex-wrap, переделал инпуты в optin-section, теперь не выходят за рамки

Yashiandr reviewed (commented):
Очень неплохо получилось, молодец, что за столь короткое время общения смог уловить суть. Теперь осталось сдлеать еще немного общих классов, так называемые миксины и еще подсократить код, у тебя например background-size  написан много раз у каждой иконки, хотя её возможно разместить отдельным классом и сократить код добавив , также можно сделать и с цветами, их поменьше, и добавить как класс тому тэгу, где они присутствуют. Старайся избегать повторения кода на не существенных местах, дисплей флекс например из класса лучше не выноссить в отдельный, чтобы понимать что этот класс в ксс гидкий, позишины тоже, а вот всякий марджины, паддинги (что ты и сделал) шрифты, размер шрифтов, цвета вот такое все незначительно лучше написать в миксины, можешь посмотреть в моей репозитории как это делается (но у меня там из фанатизма почти все выпилины отдельно)

jhamish committed:
подрезал немного

Yashiandr commented on index.html:
```diff
@@ -1,10 +1,305 @@
+	<section class='section hero-section__back-image'>
+		<div class='section__main-content'>
+			<div class='section__cart _gap-1rem _padding-top-2rem'>
+				<h1 class='section__title_hero'>
+					<span class='hero-section__span hero-section__span_grad-92deg'>Dive</span>
+					Into The Depths </h1>
+				<h1 class='section__title_hero'>Of
+					<span class='hero-section__span hero-section__span_grad-92deg'>Virtual Reality</span>
+				</h1>
```
монжно объединить в один тэг h1 с заданным max-width для переноса

jegius replied:
![image](https://github.com/jegius/ivan-edu/assets/11362702/61ad04ed-ef73-4c0f-b672-06e544eebc96)
на разрешении 2k контент не выравнивается по центру

jegius commented:
![image](https://github.com/jegius/ivan-edu/assets/11362702/eb947fd9-1d4e-426d-b87c-e0ed55bc98e1)
на больших разрешениях контент не выравнивается по центру

jegius commented on css/main.css:
```diff
@@ -0,0 +1,743 @@
+}
+
+.service-image_SIMULATION {
+    background: var(--service-SIMULATION) center top 1.5rem no-repeat, var(--service-gradient);
```
все 4 класса очень похожи стилям, вынеси совпадающиеи в отдельй класс а в модификаторах оаставь только background-image: var(--service-SELF-CARE)

jegius replied:
пройдись по коду и исправь похожие места

Jhon-Crow replied:
градиент конфликтовать с изображением будет, хз как их помирить

jegius commented on css/main.css:
```diff
@@ -0,0 +1,743 @@
+}
+
+.hero-section__span_grad-92deg {
+    background: var(--hero-span-gradient_92deg);
```
так, если тут картинка повернута на градус, то лучше так не делать, если нужно повернуть на градус есть transform rotate. Везде где есть такое же - поправь

Jhon-Crow replied:
тут направление градиента

Jhon-Crow replied:
или его тоже через ротейт делать?

jegius commented on css/main.css:
```diff
@@ -0,0 +1,743 @@
+
+.hero-section__arrow {
+    background: var(--arrow_small) center no-repeat;
+    background-size: 6.25rem 6.25rem;
```
если background-size одинаковый, занчит можно просто задать 6.25rem, поправь везде где такое есть

jegius commented on css/main.css:
```diff
@@ -0,0 +1,743 @@
+}
+
+.hero-section__back-image {
+    background: var(--hero-vector_left) left top .9rem no-repeat;
```
если ты двигал картинки - то лучше так не делать, ддвижения лучше задавать или через трансформ или через позишен, старайся никогда не хардкодить картинкой

jegius commented on css/main.css:
```diff
@@ -0,0 +1,743 @@
+    transition: transform .2s ease-in-out, box-shadow .2s ease-in-out;
+    max-width: 13.375rem;
+    max-height: 3rem;
+    box-shadow: 0 0 0 0;
```
можно просто 0, пройдись по коду проверь похожые места

jhamish committed:
поправил по комментам

jhamish committed:
переминовал .optin-section__border-ы в input-ы, добавил анимацию для line-wrapper в hero-section-е

jhamish committed:
объединил падинги и тп. вынес transition в отдельный класс

jhamish committed:
вынес фонт-вэйт 700

jegius commented on css/main.css:
```diff
@@ -0,0 +1,754 @@
+    background-size: 10.875rem
+}
+
+.tech-image-unity {
```
![image](https://github.com/jegius/ivan-edu/assets/11362702/424c3510-d7ab-448d-95dd-728cdd1af658)
так вот тут слишком переусложнил, это один блок tech-image с одной и той же шининой и высотой но с разными background

jegius commented on css/main.css:
```diff
@@ -0,0 +1,754 @@
+    padding-top: 10.25rem;
+}
+
+.tech-section__back-image {
```
тут немного ошибся с композицией, этот блок нужно сделать отдешьным элементом все таки, так как по дизайну он должен быть независим от секции
![image](https://github.com/jegius/ivan-edu/assets/11362702/d74fc61e-a18d-4ee3-8366-5d48b6f01023)

jegius reviewed (commented):
Остальное вроде ок, поправь замечания и можно идти дальше

jhamish committed:
переделал tech-image

jhamish committed:
доредактил
